worker/cluster_worker.ml

```diff
@@ -236,6 +236,13 @@ let rec maybe_prune t queue =
       (fun () ->
          Lwt_process.exec ("", [| "docker"; "system"; "prune"; "-af" |]) >>= function
          | Unix.WEXITED 0 ->
+           (match obuilder with
+           | None -> Lwt.return 0
+           | Some obuilder ->
+             match Obuilder_build.backend obuilder with
+             | `Native _ -> Lwt.return 0
+             | `Docker _ -> Obuilder_build.purge obuilder) >>= fun n ->
+           Log.info (fun f -> f "%i items prune from Obuilder" n);
            Lwt_process.exec ("", [| "docker"; "builder"; "prune"; "-af" |])
```

Comment:
I think this is out of scope for your changes, but seems worth clarifying any how:

According to https://docs.docker.com/reference/cli/docker/builder/prune/, this will

> Remove all unused build cache, not just dangling ones

Whereas, according to https://docs.docker.com/reference/cli/docker/system/prune/, `docker system prune` (which we should have just called on line 237) will

> Remove all unused containers, networks, images (both dangling and unused), and optionally, volumes.

Based on the warning in the docks, this includes all

> unused build cache

So I'm wondering whether we actually need this additional step, or if `docker system prune` should not already cover it...